src/L2SharedBridge.sol

```diff
@@ -1,183 +1,124 @@
     /// @param _l2Receiver The account address that would receive minted ether
-    /// @param _l1Token The address of the token that was locked on the L1
+    /// _l1Token The address of the token that was locked on the L1
     /// @param _amount Total amount of tokens deposited from L1
-    /// @param _data The additional data that user can pass with the deposit
-    function finalizeDeposit(
-        address _l1Sender,
-        address _l2Receiver,
-        address _l1Token,
-        uint256 _amount,
-        bytes calldata _data
-    ) external override {
+    /// _data [UNUSED] The additional data that user can pass with the deposit
+    function finalizeDeposit(address _l1Sender, address _l2Receiver, address, uint256 _amount, bytes calldata)
+        external
+        override
+    {
         // Only the L1 bridge counterpart can initiate and finalize the deposit.
-        require(
-            AddressAliasHelper.undoL1ToL2Alias(msg.sender) == l1Bridge ||
-                AddressAliasHelper.undoL1ToL2Alias(msg.sender) == l1SharedBridge,
-            "mq"
-        );
-
-        address expectedL2Token = l2TokenAddress(_l1Token);
-        address currentL1Token = l1TokenAddress[expectedL2Token];
-        if (currentL1Token == address(0)) {
-            address deployedToken = _deployL2Token(_l1Token, _data);
-            require(deployedToken == expectedL2Token, "mt");
-            l1TokenAddress[expectedL2Token] = _l1Token;
-        } else {
-            require(currentL1Token == _l1Token, "gg"); // Double check that the expected value equal to real one
-        }
-
-        IL2StandardToken(expectedL2Token).bridgeMint(_l2Receiver, _amount);
-        emit FinalizeDeposit(_l1Sender, _l2Receiver, expectedL2Token, _amount);
-    }
-
-    /// @dev Deploy and initialize the L2 token for the L1 counterpart
-    function _deployL2Token(address _l1Token, bytes calldata _data) internal returns (address) {
-        bytes32 salt = _getCreate2Salt(_l1Token);
-
-        BeaconProxy l2Token = _deployBeaconProxy(salt);
-        L2StandardERC20(address(l2Token)).bridgeInitialize(_l1Token, _data);
-
-        return address(l2Token);
+        require(undoL1ToL2Alias(msg.sender) == l1SharedBridge, "mq");
+        MintableToken(L2_USDC_TOKEN).mint(_l2Receiver, _amount);
+        emit FinalizeDeposit(_l1Sender, _l2Receiver, L2_USDC_TOKEN, _amount);
     }
 
     /// @notice Initiates a withdrawal by burning funds on the contract and sending the message to L1
     /// where tokens would be unlocked
     /// @param _l1Receiver The account address that should receive funds on L1
-    /// @param _l2Token The L2 token address which is withdrawn
+    /// _l2Token The L2 token address which is withdrawn
     /// @param _amount The total amount of tokens to be withdrawn
-    function withdraw(address _l1Receiver, address _l2Token, uint256 _amount) external override {
+    function withdraw(address _l1Receiver, address, uint256 _amount) external override {
```

Comment:
Nit: If the token is specified, then it should be validated, so that users don't screw up.

Reply:
the `_l2Token` param is just there to comply with the interface, but it's not used at all, you can send whatever address but the function will always use the USDC one